Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1925,27 +1931,28 @@ private static ErrorInformation CreateErrorInformation(Counterexample error, VC.
         }
         else
         {
-          var msg = assertError.FailingAssert.ErrorData as string;
-          var tok = assertError.FailingAssert.tok;
-          if (!CommandLineOptions.Clo.ForceBplErrors && assertError.FailingAssert.ErrorMessage != null)
+          string msg = null;
+          string bec = null;
+          if (assertError.FailingAssert.ErrorMessage == null || CommandLineOptions.Clo.ForceBplErrors)
+          {
+            msg = assertError.FailingAssert.ErrorData as string;
+            if (msg == null)
+            {
+              msg = "This assertion might not hold.";
+              bec = "BP5001";
+            }
+          }
+          else
           {
             msg = assertError.FailingAssert.ErrorMessage;
-            tok = null;
             if (cause == "Error")
```

Comment:
Can we remove this branch? I don't see the point.
It would then print `Error: ` in front of the message, which would be consistent with all other cases.
@alexanderjsummers @gauravpartha: I saw that Viper makes use of the `:msg` attribute. Would this break any of your parsing of Boogie's output?